videoclip.lua

```diff
@@ -431,6 +475,90 @@ function main_menu:update()
+    local endpoint = config.litterbox and 'https://litterbox.catbox.moe/resources/internals/api.php' or 'https://catbox.moe/user/api.php'
+
+    encoder.create_clip('video',
+        function(outfile)
+            notify("Uploading to " .. (config.litterbox and "litterbox.catbox.moe..." or "catbox.moe..."), "info", 9999)
+
+            -- This uses cURL to send a request to the cat-/litterbox API.
+            -- cURL is included on Windows 10 and up, most Linux distributions and macOS.
+
+            local r = subprocess({ -- This is technically blocking, but I don't think it has any real consequences ..?
+                os_type == 'windows' and 'curl.exe' or 'curl', '-s',
+                '-F', 'reqtype=fileupload',
+                '-F', 'time=' .. config['litterbox_expire'],
+                '-F', 'fileToUpload=@"' .. outfile .. '"',
+                endpoint
+            })
+
+            -- Exit codes in the range [0, 99] are returned by cURL itself.
+            -- Any other exit code means the shell failed to execute cURL.
+            if r.status < 0 or r.status > 99 then
+                notify("Error: Failed to upload. Make sure cURL is installed and in your PATH.", "error", 3)
+                return
+            end
+            if r.status ~= 0 then
+                notify("Error: Failed to upload to " .. config.litterbox and "litterbox.catbox.moe" or "catbox.moe", "error", 2)
+                return
+            end
+
+            -- Copy to clipboard
+            if os_type == 'windows' then
+                local clipboard_command = {
+                    'powershell.exe', '-command',
+                    'Set-Clipboard -Value ' .. r.stdout
+                }
+                
+                local cb = subprocess(clipboard_command)
+                -- local status = mp.command('run ' .. clipboard_command)
+                if cb.status ~= 0 then
+                    notify("Failed to copy URL to clipboard, trying to open in browser instead. (Make sure PowerShell is installed)", "warn", 4)
+                    mp.commandv('run', open_utility, r.stdout)
+                    return
+                end
+            end
+
+            if os_type == 'macos' then
+                local clipboard_command = {
+                    "pbcopy"
+                }
+                local cb = subprocess(clipboard_command, r.stdout)
+                if cb.status ~= 0 then
+                    notify("Failed to copy URL to clipboard, trying to open in browser instead. (Make sure pbcopy is installed)", "warn", 4)
+                    mp.commandv('run', open_utility, r.stdout)
+                    return
+                end
+            end
+
+            if os_type == "linux" then
+                local cb_x11 = string.find(subprocess({
+                    "whereis", "xclip"
+                }).stdout, "/")
+                local cb_wayland = string.find(subprocess({
+                    "whereis", "wl-copy"
+                }).stdout, "/")
```

Comment:
You can have both xclip and wl-copy installed. Better if you do:
```
if is_wayland() then
clipboard_command = "wlcopy"
else
clipboard_command = "xclip"
end
```

Reply:
It was a lot cleaner originally, but I ran into a lot of problems trying to get the Linux commands to work (see https://github.com/Ajatt-Tools/videoclip/pull/20#issuecomment-1599381497), but tl;dr there's not way to run the command synchronously, and there's no way to get the exit code asynchronously to check if the command fails. I ended up opting for the current system that checks PATH for either of the binaries first, runs one if found, or opens in browser if not.

Reply:
6c571a8